sdk/openai/openai/test/public/node/batches.spec.ts

```diff
@@ -0,0 +1,80 @@
+// Copyright (c) Microsoft Corporation.
+// Licensed under the MIT license.
+
+import { matrix } from "@azure-tools/test-utils-vitest";
+import { describe, beforeEach, it } from "vitest";
+import OpenAI, { AzureOpenAI, toFile } from "openai";
+import { createClient } from "../utils/createClient.js";
+import {
+  APIVersion,
+  DeploymentInfo,
+  getDeployments,
+  maxRetriesOption,
+  withDeployments,
+} from "../utils/utils.js";
+import { FileObject } from "openai/resources/files.mjs";
+import { assertBatch } from "../utils/asserts.js";
+
+// TODO: Unskip the test when the delete file behavior is updated
+describe.skip("Batches", () => {
+  matrix([[APIVersion.Preview]] as const, async function (apiVersion: APIVersion) {
+    describe(`[${apiVersion}] Client`, () => {
+      let client: AzureOpenAI | OpenAI;
+      let file: FileObject;
+      let deployments: DeploymentInfo[] = [];
+
+      beforeEach(async function () {
+        client = createClient(apiVersion, "completions", maxRetriesOption);
+        deployments = await getDeployments("vision");
+      });
+
+      describe("all CRUD APIs", function () {
+        async function createBatchFile(deploymentName: string) {
+          const inputObject = `{ "custom_id": "request-1", "method": "POST", "url": "/v1/chat/completions", "body": { "model": "${deploymentName}", "messages": [{ "role": "system", "content": "You are a helpful assistant." }, { "role": "user", "content": "What is 2+2?" }] } }`;
+          file = await client.files.create({
+            file: await toFile(Buffer.from(inputObject), "batch.jsonl"),
+            purpose: "batch",
+          });
+          while (file.status !== "processed") {
+            await new Promise((resolve) => setTimeout(resolve, 1000));
+            file = await client.files.retrieve(file.id);
+          }
+          return file;
```

Comment:
It is not clear to me why are we both setting the file var and returning it. Could we simplify this logic by getting rid of the global file var?

Reply:
Updated